presto-main/src/test/java/com/facebook/presto/sql/planner/plan/TestAssignments.java

```diff
@@ -13,23 +13,27 @@
  */
 package com.facebook.presto.sql.planner.plan;
 
+import com.facebook.presto.spi.plan.Assignments;
 import com.facebook.presto.spi.relation.VariableReferenceExpression;
-import com.google.common.collect.ImmutableCollection;
 import org.testng.annotations.Test;
 
+import java.util.List;
+
 import static com.facebook.presto.spi.type.BigintType.BIGINT;
 import static com.facebook.presto.sql.planner.iterative.rule.test.PlanBuilder.assignment;
 import static com.facebook.presto.sql.tree.BooleanLiteral.TRUE_LITERAL;
 import static org.testng.Assert.assertTrue;
 
-public class TestAssingments
+public class TestAssignments
 {
     private final Assignments assignments = assignment(new VariableReferenceExpression("test", BIGINT), TRUE_LITERAL);
 
-    @Test
+    @Test(expectedExceptions = {UnsupportedOperationException.class})
     public void testOutputsImmutable()
     {
-        assertTrue(assignments.getOutputs() instanceof ImmutableCollection);
```

Comment:
Rather than throwing an exception, use `UnmodifiableCollection`

Reply:
So there's no "easy" way to check if a collection is an UnmodifiableCollection since the function unmodifiableList is public however the static class UnmodifiableList is package private. So we can use java reflection to test this (as opposed to instanceof):

```
assertTrue(assignments.getOutputs().getClass().isInstance(unmodifiableList(new ArrayList<>())));
```

Which seemed a little ugly so I chose to throw an exception.

Here is an example in SPI that does a similar thing: https://github.com/prestodb/presto/blob/master/presto-spi/src/test/java/com/facebook/presto/spi/predicate/TestEquatableValueSet.java#L299

Reply:
Good point. Keep as is then.